pkg/cmd/pulumi/package_gen_sdk.go

```diff
@@ -49,53 +54,90 @@ func newGenSdkCommand() *cobra.Command {
 			}
 			if language == "all" {
 				for _, lang := range []string{"dotnet", "go", "java", "nodejs", "python"} {
-					err := genSDK(lang, out, pkg)
+					// If we're generating all the languages place each one in a subdirectory of the output
+					// directory, and grab the overlays from a subdirectory of the overlays directory.
+					langOut := filepath.Join(out, language)
+					langOverlays := ""
+					if overlays != "" {
+						langOverlays = filepath.Join(overlays, language)
+					}
+					err := genSDK(lang, langOut, pkg, langOverlays)
 					if err != nil {
 						return err
 					}
 				}
 				return nil
 			}
-			return genSDK(language, out, pkg)
+			return genSDK(language, out, pkg, overlays)
 		}),
 	}
 	cmd.Flags().StringVarP(&language, "language", "", "all",
 		"The SDK language to generate: [nodejs|python|go|dotnet|java|all]")
 	cmd.Flags().StringVarP(&out, "out", "o", "./sdk",
 		"The directory to write the SDK to")
+	cmd.Flags().StringVar(&overlays, "overlays", "", "A folder of extra overlay files to copy to the generated SDK")
+	contract.AssertNoErrorf(cmd.Flags().MarkHidden("overlays"), `Could not mark "overlay" as hidden`)
 	return cmd
 }
 
-func genSDK(language, out string, pkg *schema.Package) error {
+func genSDK(language, out string, pkg *schema.Package, overlays string) error {
 	var f func(string, *schema.Package, map[string][]byte) (map[string][]byte, error)
 	switch language {
 	case "csharp", "c#", "dotnet":
 		f = dotnet.GeneratePackage
-		language = "dotnet"
 	case "go":
+		if overlays != "" {
+			return errors.New("overlays are not supported for Go")
+		}
 		f = func(s string, p *schema.Package, m map[string][]byte) (map[string][]byte, error) {
 			return gogen.GeneratePackage(s, pkg)
 		}
 	case "typescript", "nodejs":
 		f = nodejs.GeneratePackage
-		language = "nodejs"
 	case "python": //nolint:goconst
 		f = python.GeneratePackage
 	case "java":
 		f = javagen.GeneratePackage
 	}
 
-	m, err := f("pulumi", pkg, nil)
+	extraFiles := make(map[string][]byte)
+	if overlays != "" {
+		err := filepath.WalkDir(overlays, func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if d.IsDir() {
+				return nil
+			}
+
+			contents, err := os.ReadFile(path)
+			if err != nil {
+				return fmt.Errorf("could not read overlay file %s: %w", path, err)
+			}
+
+			relative, err := filepath.Rel(overlays, path)
+			contract.AssertNoError(err)
+
+			extraFiles[relative] = contents
+
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("could not read overlay directory %s: %w", overlays, err)
```

Comment:
optional nit/FYI:
It's recommended not to use "could not foo" or "error doing foo" form for strings in error objects. Just "foo".
The reasoning for this is that these are typically chained together so:

    could not build overlay: could not read file: could not open "bar": does not exist

Versus

    build overlay: read file: open "bar": does not exist

(This does not apply to UI boundaries like log messages and messages printed to the user. Those should be normal sentences as usual.)

That said, conformance to this is mixed across our codebase (and even in the wider community) so 🤷.